nixos/modules/config/networking.nix

```diff
@@ -32,6 +32,22 @@ in
       '';
     };
 
+    networking.addFQDNAndHostnameEntries = lib.mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether to add the FQDN and hostname entries to /etc/hosts,
+        mapping FQDN and hostname to 127.0.0.2 and optionally ::1 if
+        IPv6 is enabled.
+
+        Setting this to false prevents DNS entries from being
+        overridden (when FQDN should resolve to the public IP address
+        instead of 127.0.0.1)
+
+        Useful for e.g. `services.zookeeper`.
```

Comment:
Maybe include this note in the zookeeper options, and not here?

Reply:
Yeah, that's better ec5c2b22b2a6dbf9047f8000f2aa014247dfd2db